### ISSUE TITLE
go-vcr messes with a Body set to http.NoBody

### ISSUE THREAD

**Issue by vangent** [dnaeon/go-vcr#38]
According to `net/http` (https://golang.org/pkg/net/http/):

```
http.NoBody can be used in an outgoing client request to explicitly signal that a request has zero bytes. An alternative, however, is to simply set Request.Body to nil.
```

[recorder.go](https://github.com/dnaeon/go-vcr/blob/master/recorder/recorder.go#L99) checks for `r.Body != nil` before inserting a new `Reader` (a `TeeReader` used to save the Body into a buffer in order to record it). However, if `r.Body` is `http.NoBody`, it still inserts the new `Reader`. This can result in the HTTP request being made without a `Content-Length=0` header.

**Comment by vangent** [comment]
https://github.com/golang/go/issues/20257#issuecomment-299509391 is relevant.

**Comment by vangent** [comment]
Fixed with #39.
